runtime/jvmti/jvmtiRawMonitor.c

```diff
@@ -259,9 +259,43 @@ jvmtiRawMonitorWait(jvmtiEnv *env,
 			break;
 		case J9THREAD_INTERRUPTED:
 			/* FALLTHROUGH*/
-		case J9THREAD_PRIORITY_INTERRUPTED:
+		case J9THREAD_PRIORITY_INTERRUPTED: {
+#if JAVA_SPEC_VERSION >= 19
+			J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+			j9object_t threadObject = NULL;
+			vmFuncs->internalEnterVMFromJNI(currentThread);
+			threadObject = currentThread->threadObject;
+			Assert_JVMTI_true(NULL != threadObject);
+			j9object_t threadLock = J9VMJAVALANGTHREAD_LOCK(currentThread, threadObject);
+			threadLock = (j9object_t)vmFuncs->objectMonitorEnter(currentThread, threadLock);
+			if (J9_OBJECT_MONITOR_ENTER_FAILED(threadLock)) {
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+				if (J9_OBJECT_MONITOR_CRIU_SINGLE_THREAD_MODE_THROW == (UDATA)threadLock) {
+					vmFuncs->setCRIUSingleThreadModeJVMCRIUException(currentThread, 0, 0);
+					rc = JVMTI_ERROR_THREAD_SUSPENDED;
+				} else
+#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
+				if (J9_OBJECT_MONITOR_OOM == (UDATA)threadLock) {
+					rc = JVMTI_ERROR_OUT_OF_MEMORY;
+				}
+				goto release;
+			} else {
+				/* Refetch J9VMThread->threadObject as it can be moved by the GC while waiting for the monitor. */
+				threadObject = currentThread->threadObject;
+				if (J9VMJAVALANGTHREAD_DEADINTERRUPT(currentThread, threadObject)) {
+					J9VMJAVALANGTHREAD_SET_DEADINTERRUPT(currentThread, threadObject, JNI_FALSE);
+					if (IS_JAVA_LANG_VIRTUALTHREAD(currentThread, threadObject)) {
+						J9VMJAVALANGTHREAD_SET_DEADINTERRUPT(currentThread, currentThread->carrierThreadObject, JNI_FALSE);
+					}
+				}
+				vmFuncs->objectMonitorExit(currentThread, threadLock);
```

Comment:
This should be functionally equivalent to `Thread.getAndClearInterrupt()`, so Im not sure force setting `Thread.interrupted` to false have the correct behaviour, should the state be checked first before clearing?

Reply:
`Thread.interrupted` is only being cleared if its old value is true, similar to [Thread.getAndClearInterrupt](https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/8c7b730b4190755fa092c2b0b8bde70d9c11157f/src/java.base/share/classes/java/lang/Thread.java#L1797). Can you provide more insights on how the `state` should be checked?

Reply:
I was looking at https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/src/java.base/share/classes/java/lang/VirtualThread.java#L919
wasn't sure if the sync lock should be before the read or after. I guess current code matches the format in j.l.Thread.